portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,43 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/* Servlet that returns json of commentaries statistics*/
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
+  /* Some hardcoded examples*/
+  String [] commentList = {"Hi! I'm Mark", "Hello, my name is Nancy :3", "Good morning/afternoon/evening! I'm Alex"};
+  String [] nameList = {"Mark", "Nancy", "Alex"};
+  String [] ratingList = {"4/5", "5/5", "0/5"};
+  int listLength = commentList.length;
+
+  /* Processing of GET request to /comment page, json dtring will be returned*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    String json = "{";
```

Comment:
Concatenating strings multiple times in java is inefficient. Please use a StringBuilder to append the values if it is really needed and convert to a string at the end. (https://docs.oracle.com/javase/tutorial/java/data/buffers.html)

In this case, it is not needed. As per the comment below, all conversion to JSON format should happend with gson. This means you should build a `List` of `Comment` objects and then use gson to convert the entire list to JSON in one go and write it out.